src/profile-logic/merge-compare.js

```diff
@@ -916,6 +920,7 @@ function combineSamplesDiffing(
       // of eventDelay/responsiveness don't mean anything.
       newSamples.eventDelay.push(null);
       newSamples.time.push(samples2.time[j]);
+      newThreadId.push(samples2.threadId ? samples2.threadId[j] : tid2);
```

Comment:
In addition to using them in the CPU usage calculation, we can also add them to sample tooltips. We were discussing about adding a track for each merged track, which means that we are also going to see the merged thread in the activity graph. But we don't have this feature yet.